src/NuGet.Server/Infrastructure/ServerPackageRepository.cs

```diff
@@ -458,17 +458,25 @@ private HashSet<ServerPackage> ReadPackagesFromDisk()
 
                 Parallel.ForEach(packages, package =>
                 {
-                    // Create server package
-                    var serverPackage = CreateServerPackage(package, enableDelisting);
-
-                    // Add the package to the cache, it should not exist already
-                    if (cachedPackages.Contains(serverPackage))
+                    try
                     {
-                        _logger.Log(LogLevel.Warning, "Duplicate package found - {0} {1}", package.Id, package.Version);
+                        // Create server package
+                        var serverPackage = CreateServerPackage(package, enableDelisting);
+
+                        // Add the package to the cache, it should not exist already
```

Comment:
You can either:

1. Catch just around CreateServerPackage, something like:

```c#
ServerPackage foo;

try
{
 foo = CreateServerPackage(xxx, yyy);
}
catch
{
}

if (foo != null)
{
...
```

or go with the tryCreate 

I prefer the latter, but I'll let @maartenba make a final call